src/main/drivers/serial_uart.c

```diff
@@ -72,17 +73,10 @@ static void uartReconfigure(uartPort_t *uartPort)
 
     USART_InitStructure.USART_BaudRate = uartPort->port.baudRate;
     USART_InitStructure.USART_WordLength = USART_WordLength_8b;
-    if (uartPort->port.mode & MODE_SBUS) {
-        USART_InitStructure.USART_StopBits = USART_StopBits_2;
-        USART_InitStructure.USART_Parity = USART_Parity_Even;
-    } else {
-        if (uartPort->port.mode & MODE_STOPBITS2)
-            USART_InitStructure.USART_StopBits = USART_StopBits_2;
-        else
-            USART_InitStructure.USART_StopBits = USART_StopBits_1;
 
-        USART_InitStructure.USART_Parity = USART_Parity_No;
-    }
+    USART_InitStructure.USART_StopBits = (uartPort->port.options & SERIAL_STOPBITS_2) ? USART_StopBits_2 : USART_StopBits_1;
```

Comment:
You need explicit mask for multibits fields and using and declaring masks everywhere is a bit cumbersome.
`(uartPort->port.options & SERIAL_NOT_INVERTED) == SERIAL_NOT_INVERTED` is possible mistake that take a while to find ... 

 Using bitfields in struct may be an option (nice bonus is that value is easily visible in debugger), but gcc won't probably pack it better than sizeof(int). attribute __packed__ may help.